### PR TITLE
Address Endpoints, Tests, Schema

### LINKED ISSUE #25: Create Read on Address
See [Database Schema Doc](https://github.com/uoftblueprint/tces/wiki/Database-Schema) for info.

API Endpoint info:
- GET `/api/address/{address_id}/`. No headers, response should be JSON payload of address with id `address_id`. 404 if not found.
- POST `/api/address`. Expects to receive all the fields for a new `address` object in the request body.

This should be split up into:
* a class in Model directory (Handle interacting with MongoDB directly)
* a class in Controller directory (This will use your class in Model directory to perform the expected functionality for each endpoint)
* a class in Routes directory (This will use functions from the class you define in the Controller directory, but actually tie them to the endpoint)

See [project structure
](https://github.com/uoftblueprint/tces/wiki/Project-Structure) for further reference. Also msg me for clarification, I think this is a pretty confusing concept.

Relies on #14.

### PR DESCRIPTION
**Pull request by EclipseIsDead**
# Description

4 endpoints:
- Get address + Create address
- Update + Delete

Unit tests (integrated), schema

Address -> 
    street_address
    city
    building_num
    province
    postal_code

Fixes #25, #50 

## Sample curl
Ignore if PR does not affect an endpoint.
## Response
Ignore if PR does not affect an endpoint.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] I have run `npm run lint:fix`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works (Delete if frontend)

### CODE REVIEW

**Comment by kenneth-miura** [comment]
i am style reviewer, plz add 2 others still

File: server/src/database/schemas.ts [hunk 1]
```diff
@@ -53,4 +53,37 @@ export const jobSchema = {
...
+      },
+      street_address: {
+        bsonType: "string",
+        description: "'street_address' is required and must be a String"
+      },
+      city: {
+        bsonType: "string",
+        description: "'city' is required and must be a String"
+      },
+      building_num: {
```
Comment (helen-li) [inline server/src/database/schemas.ts:75]: schema tells me `building_num` is numeric?

Comment (helen-li) [inline server/src/database/schemas.ts:75]: schema tells me `building_num` is numeric?

File: server/src/models/address.model.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import { IntegerType, ObjectId } from "mongodb";
+import {  collections } from "../database/conn";
+import { deleteAddress } from "../controllers/address.controller";
```
Comment (helen-li) [inline server/src/models/address.model.ts:3]: do you need this import?

File: server/src/routes/address.routes.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import express, { Request, Response } from "express";
+import { ObjectId } from "mongodb";
+import Address from "../models/address.model";
+import * as AddressController from "../controllers/address.controller";
+
+export const addressRouter = express.Router();
+addressRouter.use(express.json());
```
Comment (helen-li) [inline server/src/routes/address.routes.ts:7]: might have to add this router to `index.ts`

**Review by helen-li — COMMENTED** [review]
Looks good! Just a few quick comments

**Review by jordanjanakievski — CHANGES_REQUESTED** [review]
I left a comment under Helen's about router being added to index.ts. I think that will be needed to get the collection initialized in the database. You can copy the previous code snippets in index

**Review by kenneth-miura — APPROVED** [review]
general style looks good

File: client/src/App.tsx [hunk 4]
```diff
@@ -24,3 +50,4 @@ function App() {
 }
 
 export default App;
+>>>>>>> d1222a4756d437816347512cab1d7ba9287f4c2c
```
Comment (helen-li) [inline client/src/App.tsx:53]: lol merge conflict be like

File: client/src/index.tsx [hunk 5]
```diff
@@ -1,3 +1,17 @@
+<<<<<<< HEAD
```
Comment (helen-li) [inline client/src/index.tsx:1]: same thing with merge here

File: server/src/database/conn.ts [hunk 6]
```diff
@@ -1,5 +1,8 @@
 import * as mongoDB from "mongodb";
 import * as dotenv from "dotenv";
+<<<<<<< HEAD
```
Comment (helen-li) [inline server/src/database/conn.ts:3]: a bunch here as well

**Review by helen-li — COMMENTED** [review]
slay, just a few quick necessary fixes with merge conflicts but otherwise looks great

**Review by jordanjanakievski — APPROVED** [review]
My previous comment was addressed so it looks good to me. Once those merge conflicts are handled that are mentioned by Helen, it will be good to go.

**Review by helen-li — APPROVED** [review]
looks greattt

**Pull request merged**